hnn_core/cell.py

```diff
@@ -293,27 +307,34 @@ def move_to_pos(self):
     # 2. a list needs to be created with a Dipole (Point Process) in each
     #    section at position 1
     # In Cell() and not Pyr() for future possibilities
-    def insert_dipole(self, yscale):
+    def insert_dipole(self, p_secs, dipole_sec):
         """Insert dipole into each section of this cell.
 
         Parameters
         ----------
-        yscale : dict
-            Dictionary of length scales to calculate dipole without
-            3d shape.
+        p_secs : nested dict
+            Dictionary with keys as section name.
+            p_secs[sec_name] is a dictionary with keys
+            L, diam, Ra, cm, syns and mech.
+            syns is a list specifying the synapses at that section.
+            The properties of syn are specified in p_syn.
+            mech is a dict with keys as the mechanism names. The
+            values are dictionaries with properties of the mechanism.
+        dipole_sec : str
```

Comment:
Along the lines of my other comments, could this be `'apical_section'`?

`The name of a section defining the apical orientation, onto which dipole moments are projected.`

Reply:
isn't "apical" specific to the default neuron? I like the use of term "projection" though. I'll try to incorporate it

Reply:
In my mind, "apical" refers to the single large dendrite sprouting from the top of the pyramid-formed soma (from the "apex" of the pyramid). Some L2/3 pyramidal cells have a bifurcation very close to the soma, but even then the two apical dendrites run in parallel, towards L1. The "textbook answer" is that the apical dendrite(s) is a defining characteristic of _all_ neocortical (indeed, also hippocampal) pyramidal cells. The terms 'proximal' and 'distal' are much more ambiguous in comparison.

Reply:
humm ... so are you saying that you'd always want to compute the dipole along an apical dendrite? I'll change the name in that case

Reply:
Yes. Even if we one day included a more morphologically detailed model, the dipole moment would still be calculated along the primary symmetry axis defined by the apical dendrite. You know how in MNE we have the "loose orientation constraint"? The "looseness" is fudge factor to cover for coreg and surface recon errors. The underlying _strong_ assumption is that there's one dominant dipole orientation per source space point. And this is the normal to the cortical (WM) surface _because of_ the apical dendrites.